test/src/meshgenerators/TestMeshGenerator.C

```diff
@@ -0,0 +1,138 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "TestMeshGenerator.h"
+
+registerMooseObject("MooseTestApp", TestMeshGenerator);
+
+InputParameters
+TestMeshGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+
+  params.addParam<MeshGeneratorName>("input", "Input mesh for testing");
+  params.addParam<std::vector<MeshGeneratorName>>("inputs", "Input meshes for testing");
+  params.addParam<MeshGeneratorName>("invalid_input", "Input that isn't set");
+
+  params.addParam<bool>("dummy", false, "Dummy parameter");
+
+  params.addParam<bool>("get_mesh_invalid", false, "Tests the getMesh invalid parameter error");
+  params.addParam<bool>("get_mesh_wrong_type", false, "Tests the getMesh wrong type error");
+  params.addParam<bool>(
+      "get_meshes_by_name_invalid", false, "Tests the getMeshesByName invalid parameter error");
+  params.addParam<bool>(
+      "get_meshes_by_name_wrong_type", false, "Tests the getMeshesByName wrong type error");
+  params.addParam<bool>(
+      "get_mesh_outside_construct", false, "Tests getting a mesh outside of construction");
+  params.addParam<bool>(
+      "declare_mesh_prop_outside_construct", false, "Declare a mesh property out of construction");
+  params.addParam<bool>(
+      "sub_outside_construct", false, "Tests building a subgenerator outside of construction");
+  params.addParam<std::string>("null_mesh_name", "Declares a null mesh with this name");
+  params.addParam<bool>(
+      "mesh_prop_double_declare", false, "Tests declaring the same mesh property twice");
+  params.addParam<bool>("sub_no_declare_input",
+                        false,
+                        "Tests getting an input from a sub generator but not declaring it");
+  params.addParam<bool>("input_not_moved", false, "Tests when an input mesh is not moved");
+  params.addParam<bool>("request_input", false, "Request the mesh in 'input'");
+  params.addParam<bool>("request_inputs", false, "Request the meshes in 'input'");
+  params.addParam<bool>("request_input_by_name", false, "Request the mesh in 'input' by name");
+  params.addParam<bool>("request_inputs_by_name", false, "Request the meshes in 'inputs' by name");
+  params.addParam<bool>("request_sub_input", false, "Request the mesh in 'input' for a sub");
+  params.addParam<bool>("missing_get", false, "Tests getting a missing meshgenerator from the app");
+  params.addParam<std::string>("add_sub_input", "Adds this input as an input to a sub generator");
+
+  return params;
+}
+
+TestMeshGenerator::TestMeshGenerator(const InputParameters & parameters) : MeshGenerator(parameters)
+{
+  if (getParam<bool>("get_mesh_invalid"))
+    static_cast<void>(getMesh("invalid"));
```

Comment:
are these casts because of the [[nodiscard]]

Reply:
yep